src/index.ts

```diff
@@ -139,8 +139,15 @@ const getWorksheet = (jsonSheet: IJsonSheet, settings: ISettings): WorkSheet =>
 const writeWorkbook = (workbook: WorkBook, settings: ISettings = {}): Buffer | undefined => {
   const filename = `${settings.fileName ?? "Spreadsheet"}.xlsx`
   const writeOptions = settings.writeOptions ?? {}
+  
+  if(settings.writeMode === 'write') {
+    return write(workbook, writeOptions)
+  } else if(settings.writeMode === 'writeFile') {
+    return writeFile(workbook, filename, writeOptions)
+  } else {
+    return writeOptions.type === "buffer" ? write(workbook, writeOptions) : writeFile(workbook, filename, writeOptions)
+  }
 
```

Comment:
If you are returning you don't need `else` in each case

Reply:
I left `else` for backward compatibility